сортировка

zayatnik opened the pull request:
@PanovaElena
это моя быстрая сортировка
там есть вопрос у меня
всё нормально сортируется, когда я иду по шагам, в конце выполнения функции имеется отсортированный вектор, однако когда выполнение функции сортировки заканчивается, вектор снова имеет первоначальный вид
я вот не понимаю почему так
пытался сделать функцию типа не void, а vector<int>, тогда естественно возвращает изменённый вектор, но не до конца отсортированный, потому что функция рекурсивна и возвращается то, что имеется после первого пробега по функции, я не придумал как описать условие, при котором возвращался бы именно последний вариант
пытался с помощью булевской переменной как-то статус запоминать, но ни к чему хорошему не пришёл

zayatnik committed:
Update vectsotr.cpp

PanovaElena commented:
Разобрались ведь с проблемой? Можете проводить эксперименты. Нужно 2 графика: в худшем случае и в среднем. Помните ли, как правильно делать запуски?

zayatnik commented:
Как их правильно делать?

zayatnik commented:
меня просто не было на той паре, я узнаю у ребят

PanovaElena commented:
Если кратко, то:

1) Нужен график зависимости времени сортировки от N.
2) Сборка Release.
3) Одна сортировка одного массива - один запуск программы.
4) Для одного N несколько запусков (не менее 3-х, результат - среднее или минимум). В случае рандомизированной быстрой сортировки нужно еще больше запусков, т.к. там сложность вычисляется в смысле мат.ожидания.
5) Если быстрая сортировка не рандомизированная (ваш случай, кажется), то нужно 2 графика: в худшем случае и в среднем.
6) Кривые на графиках должны иметь рост, соответствующий оценке сложности для сортировки. Чтобы это проверить, нужно поделить время на сложность и построить график, должна получиться константа.

zayatnik committed:
это я пока ту ошибку испарвил

zayatnik commented:
Рандомизированная - это когда опорный элемент рандомно выбирается?

zayatnik committed:
randomized

zayatnik commented:
если да, то пусть она лучше будет таковой)

zayatnik committed:
поправочка

zayatnik commented:
![image](https://user-images.githubusercontent.com/55553752/70529175-f9cd1980-1b60-11ea-88ce-3998b0dbc2b7.png)
график времени от N
для рандомизированной ведь только один график нужен, верно?

PanovaElena commented:
Добрый день! Да, нужен один такой график.   1) Не понятно, какие единицы измерения времени2) N=25000 для сортировок - маловато. Нужно бы взять хотя бы 100000.3) Непонятно, что это за функция. Чтобы удостовериться, что это n*log(n), поделите время на n*log(n) и постройте график. Должна получиться константа. 10.12.2019, 15:24, "zayatnik" <notifications@github.com>:график времени от Nдля рандомизированной ведь только один график нужен, верно?—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.  -- С уважением,Елена Панова 

zayatnik committed:
скобочки

поставил скобочки, сортировка стала просто в ДЕСЯТКИ ТЫСЯЧ раз быстрее, сейчас переделаю табличку и график

zayatnik commented:
![image](https://user-images.githubusercontent.com/55553752/70935211-b8e96f00-2050-11ea-80ac-76e3bbcc7f9c.png)
получилось почему-то вот так, практически линейная зависимость...

PanovaElena commented:
У вас получилось очень маленькое время. Нужно взять побольше n, чтобы время было хотя бы около 1 секунды. Возможно, это решит проблему.
